letsencrypt-apache/letsencrypt_apache/configurator.py

```diff
@@ -895,9 +895,10 @@ def _add_name_vhost_if_necessary(self, vhost):
         for addr in vhost.addrs:
             # In Apache 2.2, when a NameVirtualHost directive is not
             # set, "*" and "_default_" will conflict when sharing a port
+            addrs = set((addr,))
```

Comment:
Almost.

> In addition to ensuring that addrs is always defined, this change will cause add_name_vhost to be called once for every other vhost with the same address **if the address does not already have a name vhost directive.**

> Previously it would only be called if there was a "*" or "_default_" vhost.
Yes, this was also the case before the last commit on #2314, including in `master` before the PR was merged.